com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/util/ApplicationArchiveExtractor.java

```diff
@@ -24,26 +37,44 @@
     private String moduleFileName;
     private long maxSizeInBytes;
     private long currentSizeInBytes;
+    private MessageDigest applicationDigest;
+    private Set<String> knownFileNames;
 
-    public ApplicationArchiveExtractor(InputStream inputStream, String moduleFileName, long maxSizeInBytes, StepLogger logger) {
+    public ApplicationArchiveExtractor(InputStream inputStream, String moduleFileName, long maxSizeInBytes, Set<String> knownFileNames,
+        StepLogger logger) {
         this.inputStream = new ZipInputStream(inputStream);
         this.moduleFileName = moduleFileName;
         this.maxSizeInBytes = maxSizeInBytes;
+        this.knownFileNames = knownFileNames;
         this.logger = logger;
     }
 
-    public Path extract() {
-        Path appPath = null;
+    public Collection<CloudResource> getApplicationMetaData() {
```

Comment:
It is strange to me. The `getApplicationMetadata` is returning Collection of CloudResource. Is this really a metdata?